### ISSUE TITLE
CRASH:     (monitor.ml.Error (Failure "Digit.split index out of bounds")

### ISSUE THREAD

**Issue by jkrauska** [CodaProtocol/coda#3143]
Crash on several nodes on testnet:
```
    (monitor.ml.Error (Failure "Digit.split index out of bounds")
  ("Raised at file \"stdlib.ml\", line 33, characters 17-33"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 195, characters 26-46"
    "Called from file \"lib/network_pool/f_sequence.ml\" (inlined), line 328, characters 32-77"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 580, characters 12-113"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 576, characters 28-74"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 640, characters 10-36"
    "Called from file \"lib/network_pool/indexed_pool.ml\", line 532, characters 10-138"
    "Called from file \"lib/network_pool/transaction_pool.ml\", line 384, characters 30-193"
    "Called from file \"lib/network_pool/network_pool_base.ml\", line 25, characters 15-65"
    "Called from file \"src/pipe.ml\", line 827, characters 4-56"
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
    "Called from file \"src/job_queue.ml\", line 170, characters 6-47"
    "Caught by monitor coda"))
```

**Comment by jkrauska** [comment]
Happened later in testnet breakdown.

https://s3-us-west-2.amazonaws.com/crashlogs.o1test.net/testnet_share_the_wealth/2019-08-08/crashlog-digit-split.tar.bz2

**Comment by imeckler** [comment]
I looked into this one -- couldn't get it to reproduce. `F_sequence.split_at` is throwing. I tried to get it to reproduce by

- applying a transaction pool diff with 400 (> the max size of 300) transactions from one account
- calling split_at with invalid indices

and couldn't.

@enolan could you investigate?

**Comment by garethtdavies** [comment]
I just had this crash on the reinstated testnet:

```
(monitor.ml.Error (Failure "Digit.split index out of bounds")
  ("Raised at file \"stdlib.ml\", line 33, characters 17-33"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 195, characters 26-46"
    "Called from file \"lib/network_pool/f_sequence.ml\" (inlined), line 328, characters 32-77"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 580, characters 12-113"
    "Called from file \"lib/network_pool/f_sequence.ml\", line 640, characters 10-36"
    "Called from file \"lib/network_pool/indexed_pool.ml\", line 532, characters 10-138"
    "Called from file \"lib/network_pool/transaction_pool.ml\", line 384, characters 30-193"
    "Called from file \"lib/network_pool/network_pool_base.ml\", line 25, characters 15-65"
    "Called from file \"src/pipe.ml\", line 827, characters 4-56"
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
    "Called from file \"src/job_queue.ml\", line 170, characters 6-47"
    "Caught by monitor coda"))
```

**Comment by yourbuddyconner** [comment]
I have seen this crash manifest 3-4 times on the faucet daemon overnight: 

```
2019-08-09 12:05:15 UTC [Info] Removing peer from peer set: [host : 116.203.20.122, discovery_port : 8303, communication_port : 8302]
2019-08-09 12:05:15 UTC [Info] Moving peer to disconnected peer set : ((host 116.203.20.122) (discovery_port 8303) (communication_port 8302))
☠ Coda Daemon crashed. The Coda Protocol developers would like to know why!
Please:
Open an issue:
https://github.com/CodaProtocol/coda/issues/new
Briefly describe what you were doing, and include the last 20 lines from .coda-config/coda.log.
And then paste the following:
(monitor.ml.Error (Failure "Digit.split index out of bounds")
("Raised at file \"stdlib.ml\", line 33, characters 17-33"
"Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
"Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
"Called from file \"lib/network_pool/f_sequence.ml\", line 195, characters 26-46"
"Called from file \"lib/network_pool/f_sequence.ml\" (inlined), line 328, characters 32-77"
"Called from file \"lib/network_pool/f_sequence.ml\", line 580, characters 12-113"
"Called from file \"lib/network_pool/f_sequence.ml\", line 640, characters 10-36"
"Called from file \"lib/network_pool/indexed_pool.ml\", line 532, characters 10-138"
"Called from file \"lib/network_pool/transaction_pool.ml\", line 384, characters 30-193"
"Called from file \"lib/network_pool/network_pool_base.ml\", line 25, characters 15-65"
"Called from file \"src/pipe.ml\", line 827, characters 4-56"
"Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
"Called from file \"src/job_queue.ml\", line 170, characters 6-47"
"Caught by monitor coda"))
```

```
☠ Coda Daemon crashed. The Coda Protocol developers would like to know why!
Please:
Open an issue:
https://github.com/CodaProtocol/coda/issues/new
Briefly describe what you were doing, and include the last 20 lines from .coda-config/coda.log.
And then paste the following:
(monitor.ml.Error (Failure "Digit.split index out of bounds")
("Raised at file \"stdlib.ml\", line 33, characters 17-33"
"Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
"Called from file \"lib/network_pool/f_sequence.ml\", line 173, characters 18-58"
"Called from file \"lib/network_pool/f_sequence.ml\", line 195, characters 26-46"
"Called from file \"lib/network_pool/f_sequence.ml\" (inlined), line 328, characters 32-77"
"Called from file \"lib/network_pool/f_sequence.ml\", line 580, characters 12-113"
"Called from file \"lib/network_pool/f_sequence.ml\", line 576, characters 28-74"
"Called from file \"lib/network_pool/f_sequence.ml\", line 640, characters 10-36"
"Called from file \"lib/network_pool/indexed_pool.ml\", line 532, characters 10-138"
"Called from file \"lib/network_pool/transaction_pool.ml\", line 384, characters 30-193"
"Called from file \"lib/network_pool/network_pool_base.ml\", line 25, characters 15-65"
"Called from file \"src/pipe.ml\", line 827, characters 4-56"
"Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
"Called from file \"src/job_queue.ml\", line 170, characters 6-47"
"Caught by monitor coda"))
```

**Comment by imeckler** [comment]
@enolan I am reassigning this to you -- I have reproduced this, on the finger-tree-debug branch but am having trouble understanding the code where the error is happening. @bkase and @jkrauska say they are not going to do a redeploy of this network so let's do this

1. Next week if you are in can you look at this?
2. If it doesn't get fixed by the end of the week, we'll replace the finger tree with a list (the transaction pool is really small anyway, so the finger tree is not really necessary at the moment.)

**Comment by enolan** [comment]
sounds good

**Comment by enolan** [comment]
should be fixed by #3199
